scripts/pre_commit_linter.py

```diff
@@ -881,8 +883,84 @@ def _check_html_directive_name(all_files):
     return summary_messages
 
 
+def _check_directive_scope(all_files):
+    """This function checks that all directives have an explicit
+    scope key and it should not be scope: true.
+    """
+    print 'Starting directive scope check'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.js')]
+    failed = False
+    directive_does_not_have_scope = True
+    summary_messages = []
+    parser = PyJsParser()
+    for filename in files_to_check:
+        with open(filename) as f:
+            content = f.read()
+        parsed_nodes = parser.parse(content)['body']
+        for parsed_node in parsed_nodes:
+            if parsed_node['type'] == u'ExpressionStatement':
+                expression = parsed_node['expression']
+                if expression['type'] == u'CallExpression' and (
+                        expression['callee']['type'] == (
+                            u'MemberExpression')) and (
+                                expression['callee']['property']['name'] == (
+                                    u'directive')):
+                    arguments = expression['arguments']
+                    directive_name = str(arguments[0]['value']) if (
+                        arguments[0]['type'] == u'Literal') else (
+                            "No such directive exists.")
+                    arguments = arguments[1:]
+                    for argument in arguments:
```

Comment:
Way too much nesting -- makes this hard to read/comprehend. Try a pattern that's something like this:

```
if parsed_node['type'] != 'ExpressionStatement':
    continue

if ....:
    continue

etc.
```


Also I don't think you need any of the 'u' prefixes.

Reply:
Done.